ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS/DNSActions.hs

```diff
@@ -313,83 +362,190 @@ ioDNSActions =
+                       -> DNS.Domain
+                       -> (   DNS.Domain
+                           -> DNS.TYPE
+                       -> m (Maybe (Either DNSError DNSMessage)))
+                       -> StdGen
+                       -> m (DNSLookupResult peerAddr)
+srvRecordLookupWithTTL ofType tracer toPeerAddr peerType domainSRV resolveDNS rng = do
+    reply <- resolveDNS domainSRV DNS.SRV
+    case reply of
+      Nothing          -> do
+        traceWith tracer $ DnsTraceLookupError peerType Nothing domainSRV DNS.TimeoutExpired
+        return . Left $ [DNS.TimeoutExpired]
+      Just (Left  err) -> do
+        traceWith tracer $ DnsTraceLookupError peerType Nothing domainSRV err
+        return . Left $ [err]
+      Just (Right msg) ->
+        case DNS.fromDNSMessage msg selectSRV of
+          Left err -> do
+            traceWith tracer $ DnsTraceLookupError peerType Nothing domainSRV err
+            return . Left $ [err]
+          Right services -> do
+            let srvByPriority = sortOn priority services
+                grouped       = NE.groupWith priority srvByPriority
+            (result, domain) <- do
+              case listToMaybe grouped of
+                Just topPriority ->
+                  case topPriority of
+                    (domain, _, _, port) NE.:| [] -> do -- fast path
+                      result <- domainLookupWithTTL tracer ofType domain peerType resolveDNS
+                      let result' = ipsttlsWithPort port <$> result
+                      return (result', domain)
+                    many -> -- general path
+                      runWeightedLookup many
+                Nothing -> return (Right [], "")
+            case result of
+              Left {} -> traceWith tracer $ DnsSRVFail peerType domainSRV
+              Right ipsttls ->
+                traceWith tracer $ DnsResult peerType domain (Just domainSRV) ipsttls
+            return $ map (\(ip, port, ttl) -> (toPeerAddr ip port, ttl)) <$> result
 
-    lookupAAAAWithTTL :: DNS.ResolvConf
-                      -> DNS.Resolver
-                      -> DNS.Domain
-                      -> IO (Either DNS.DNSError [(IP, DNS.TTL)])
-    lookupAAAAWithTTL resolvConf resolver domain = do
-        reply <- timeout (microsecondsAsIntToDiffTime
-                           $ DNS.resolvTimeout resolvConf)
-                         (DNS.lookupRaw resolver domain DNS.AAAA)
-        case reply of
-          Nothing          -> return (Left DNS.TimeoutExpired)
-          Just (Left  err) -> return (Left err)
-          Just (Right ans) -> return (DNS.fromDNSMessage ans selectAAAA)
-          --TODO: we can get the SOA TTL on NXDOMAIN here if we want to
       where
-        selectAAAA DNS.DNSMessage { DNS.answer } =
-          [ (IPv6 addr, fixupTTL ttl)
+        ipsttlsWithPort port = map (\(ip, ttl) -> (ip, fromIntegral port, ttl))
+        runWeightedLookup :: NonEmpty (DNS.Domain, Word16, Word16, Word16)
```

Comment:
```suggestion
        ipsttlsWithPort port = map (\(ip, ttl) -> (ip, fromIntegral port, TTL))
        
        runWeightedLookup :: NonEmpty (DNS.Domain, Word16, Word16, Word16)
```
We use one empty line between local definitions, two between top level ones.

https://github.com/intersectmbo/ouroboros-network/blob/08e7e31d9024f3800cc3b5e6001e6a8e72ce5158/docs/StyleGuide.md?plain=1#L514-L515